sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -174,6 +174,10 @@ def __init__(
     ) -> None:
         super(AzureKeyCredentialPolicy, self).__init__()
         self._credential = credential
+        if not hasattr(credential, "key"):
+            raise TypeError(
+                "String is not a supported credential input type. Use an instance of AzureKeyCredential."
+            )
```

Comment:
nit: we can move this block to in front of `self._credential = credential`

Reply:
done